### DISCUSSION TITLE
DateRangeFilter didn't give the correct result because of timezone difference.

### DISCUSSION THREAD

**Discussion by xemexpress** [carltongibson/django-filter#1464] [Q&A]
For example, I am in Hong Kong and now is `2021-12-14 06:46:19`.
I filtered all instances with `date=today`, but the filter didn't give me instances with `date=2021-12-14` but `date=2021-12-13` as the `now()` which is used in `DateRangeFilter` gives `2021-12-13 22:46:19`.

This is quite frustrating. Please help.
**The alternative I propose is to have `localtime()` being used instead of `now()` in `DataRangeFilter`.**
Thank you!

**Comment by carltongibson** [comment]
Hi. As per [Django's Time Zones docs](https://docs.djangoproject.com/en/4.0/topics/i18n/timezones/) you/we **want** to do all (storage and) filtering in UTC. 

So, I think the current behaviour is correct, but there's the question of how you map to/from the local value you want in the UI. This is about making sure the widget you use gives you the right value. I'd need a fuller example to be able to say more. 

I will convert this to a discussion, since I don't (yet) think it's an issue.

**Reply by xemexpress** [reply]
Hello, thanks for the follow-up. And sorry I am not yet familiar with testing so not able to write a full test/example now. But let me provide my code snippet instead (as the best I can give at the moment). Hope it helps simulate my use case?

In `settings.py`:
```
TIME_ZONE = 'Asia/Hong_Kong'
USE_TZ = True
```

In `models.py`:
```
class MyModel(models.Model):
    date = models.DateField()
```

In `filters.py`:
```
class MyFilter(FilterSet):
    date_range = DateRangeFilter(field_name='date', label='Date Range')

    class Meta:
        model = MyModel
        fields = []
```

So, to simulate my use case, you may need to

1. Create an instance of `MyModel` with `date` equal to your local date of today.
2. (In a timezone from GMT +1 to GMT +12) Wait until your local today is (still) your "local yesterday" in UTC time format (which is the date part given by now()).
3. Run the server, select `today` for `date_range` and submit.

For example, in my case, as I am in Hong Kong (GMT+8), the **local** date time now is 2021-12-17 5:40am, and the **UTC** time is 2021-12-16 9:40pm. In step 1, I create an instance with `date` equal `2021-12-17`. In step 2, now is the moment where my local date (`2021-12-17`) is one day before the UTC date (`2021-12-16`). In step 3, I select `today` in the `date_range` field and submit.


In my case, I do not see the instance I created in step 1 in the filter/search result. But I think it should appear there in the result for our users because for a user (in my example given in step 2) experiencing the local time of 2021-12-17 5:40am and searching/filtering `today` for `date` field, he/she wouldn't expect he/she is selecting the date for the UTC version date (which can be not as same as his/her local date)?

Hope the above gives enough (though not perfect, sorry) information about the issue raised. More discussions are welcomed! Thanks!

**Reply by xemexpress** [reply]
Is it a reasonable concern here actually?

**Comment by carltongibson** [comment]
It's worth looking at the related Django ticket (and the discussion there) for TZ aware widgets in the Admin https://code.djangoproject.com/ticket/29822
